### DISCUSSION TITLE
Unable to execute the test cases on specific chrome profile using the user-data-dir argument.

### DISCUSSION THREAD

**Discussion by azfarbkhr** [seleniumbase/SeleniumBase#1893] [Q&A] [answered]
Hello Community,
 
I am not able to run my test cases on a custom chrome user profile using the Seleniumbase. 

When I run the test cases, it opens the specific chrome profile twice and opens a temporary profile and run tests on temporary profile. 

I use the following command to run the test cases: 
pytest web/test_service_portal_invoices.py --user-data-dir="C://Users//hp//AppData//Local//Google//Chrome//User Data"

Please assist.

Further information; 
I have 3 chrome profiles setup on my pc. 

Video attached for reference; 
https://github.com/seleniumbase/SeleniumBase/assets/10502857/e851de7d-a1d5-4910-9e29-05b6ea6b6cbe

**Comment by mdmintz** [comment, marked as answer]
Hello @azfarbkhr, there are a few issues with using your standard Chrome's ``User Data`` dir as the same one for Selenium:
1. Permissions issues because your regular Chrome browser is already using it as you try to connect Selenium's Chrome to it.
2. Configuration issues because the configuration that Selenium needs is not the same as regular Chrome configuration.
3. Security issues because if Selenium's Chrome accessed that, it would be able to see the data from your regular Chrome.

The solution is to use a separate folder for your user_data_dir... one that is created by Selenium so that it's not the same as the user_data_dir used by your system's default Chrome browser. Set ``--user-data-dir`` to a brand new folder that only tests will use.

**Comment by azfarbkhr** [comment]
Thanks @mdmintz 
Yeah that resolved the issue. 

I simply copied the user data folder to my project directory and used the profile path with user data dir parameter. 

This executed the cases on my chrome profile and the test cases are executed much more quickly now. 

Thanks again.
